File: modules/admin-ui/src/main/resources/web/admin/common/js/security/auth/LoginRequest.ts
module api.security.auth {

    export class LoginRequest extends AuthResourceRequest<LoginResultJson, LoginResult> {

        private loginCredentials: LoginCredentials;

        constructor(loginCredentials: LoginCredentials) {
            super();
            super.setMethod("POST");
            this.loginCredentials = loginCredentials;
        }

        getParams(): Object {
            return {
                user: this.loginCredentials.getUser(),
                password: this.loginCredentials.getPassword(),
                rememberMe: this.loginCredentials.isRememberMe()
            };
        }

        getRequestPath(): api.rest.Path {
            return api.rest.Path.fromParent(super.getResourcePath(), 'login');
        }

        sendAndParse(): wemQ.Promise<LoginResult> {

            return this.send().then((response: api.rest.JsonResponse<LoginResultJson>) => {
                return new LoginResult(response.getResult());
            });
        }

    }
}

File: modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/auth/LoginRequest.java
package com.enonic.xp.admin.impl.rest.resource.auth;

import com.enonic.xp.security.IdProviderKey;

public final class LoginRequest
{
    protected boolean rememberMe;

    protected String user;

    protected String password;

    protected String idProvider;

    public String getUser()
    {
        return user;
    }

    public IdProviderKey getIdProvider()
    {
        return idProvider == null ? null : IdProviderKey.from( idProvider );
    }

    public String getPassword()
    {
        return password;
    }

    public boolean isRememberMe()
    {
        return rememberMe;
    }
}
